### PR TITLE
translate the Getting Started guide.

### PR DESCRIPTION
**Pull request by sandrospadaro**
translate the Getting Started guide.

### CODE REVIEW

**Comment by jasonblais** [comment]
@sandrospadaro Very exciting!

@justinegeffen Would you know if there's a way to try building a version of this branch that we can use to test, or would this be a better question for our build team?

**Comment by jasonblais** [comment]
@sandrospadaro would you be open to completing the Mattermost [contribution license agreement](http://www.mattermost.org/mattermost-contributor-agreement/)? Let us know if you have any questions about it :)

**Comment by justinegeffen** [comment]
> @justinegeffen Would you know if there's a way to try building a version of this branch that we can use to test, or would this be a better question for our build team?

It can definitely be built locally but that won't be useful for multiple testers. I think the build team should be able to help though.

**Comment by jasonblais** [comment]
@cpanato when a PR was submitted to https://github.com/mattermost/mattermost-developer-documentation, we used to have S3 buckets of each build so we could test and review changes more easily.

As an example, in the attached,  clicking on the "Details" had a link to S3 which had developers.mattermost.com rendered as a webpage.

image

It was a while ago (in 2018), do you remember how you set this up and how much work would it be to set up for this PR in docs repository?

**Comment by cpanato** [comment]
@jasonblais should be very straight forward, will need some setup  in the S3 to grant some permissions and add a step in the docs to deploy that and post back the link

**Comment by jasonblais** [comment]
@cpanato That's awesome! Who would you recommend can help with it?

**Comment by cpanato** [comment]
@jasonblais the build engineer team

**Comment by jasonblais** [comment]
I have pinged our build team to see if they have time to help with that. This would be a big help both for this PR and the potential Japanese docs translations PR coming: https://community-release.mattermost.com/core/pl/k8rwijs4zbyzujmwagsifephyo

@justinegeffen While I'm away, are you open to following up with the build team some time next week, unless they've already helped by then?

@sandrospadaro I wanted to give you a heads up that I'm away Mar 24 - Apr 6, so will be slow to respond, but hopefully we'll have a test environment set up while I'm away. I'm super, super excited for this :D

**Comment by justinegeffen** [comment]
Update on this: A J[ira issue](https://mattermost.atlassian.net/browse/IS-363) has been opened to set up an S3 bucket as per Carlos's feedback.

**Comment by jasonblais** [comment]
Big thanks @sandrospadaro for continuing to work on this PR :tada: 

@justinegeffen @faase  I'm wondering if there's an opportunity to set up an environment we could use to start reviewing the changes while the build team works on setting up the S3 bucket?

**Comment by jasonblais** [comment]
@sandrospadaro Really appreciate the updates and sorry for the delay in getting this set up on an environment where we can all review and test the translations experience.

@sandrospadaro Comes to mind - do you have an environment running locally that could be exposed publicly via, say, ngrok.io?

**Comment by sandrospadaro** [comment]
@jasonblais do you mean  a web server to publish the docs translated?

**Comment by jasonblais** [comment]
@sandrospadaro I was thinking of a public URL that exposes your local web server (such as via ngrok.com).

Absolutely not required, but it would help with reviews, as it may take a while before the build team has time to do so

**Comment by sandrospadaro** [comment]
@jasonblais I published the translation [here](http://mattermost-docs.sandrospadaro.tk/
)
Keep in mind that I only translated the Getting Started guide.

**Comment by jasonblais** [comment]
@sandrospadaro I just saw this - you totally made my day. Super exciting :tada: :tada: :tada: 

image

Do you have thoughts on how a user may change between languages? I've been taking a look at a few websites and noticed a few options, such as:
 - language picker in top header of site
 - language picker in bottom footer of site

**Comment by sandrospadaro** [comment]
@jasonblais you can use [readthedocs](https://readthedocs.org/) to publish the guides.

[Here](https://mattermost-docssandrospadaro.readthedocs.io/en/latest/guides/user.html) i published a proof. Lower left you can see the options to switch language.

**Comment by jasonblais** [comment]
@sandrospadaro That's great!!

Do you know if the language option can be added to the existing picker we have in lower left corner of docs.mattermost.com? We could then have two sections - one for the language, second for the contribute section.

image

I did notice the new picker on your test server adds an advertisement in the left navigation, not sure if it's only for me?

image

**Comment by sandrospadaro** [comment]
@jasonblais I believe the options depend on Read The Docs. The "Edit on Github" option is already present, but I don't know how to add the "Report a problem" option.

About advertising, you can remove it by becoming a Read The Docs "Gold Member".
Screenshot from 2020-04-24 22-46-01

**Comment by jasonblais** [comment]
@sandrospadaro Excellent to know. Sorry for the delay, I have it on my list to review what is left for us to release first version of Italian docs translations :) :tada: 

Anything you have in mind that we should consider before a first release?

**Comment by sandrospadaro** [comment]
@jasonblais Nothing at the moment. Many thanks

File: README.md [hunk 1]
```diff
@@ -55,3 +55,15 @@ If you've downloaded the repo and are editing Mattermost documentation on your l
 6. Navigate to the `/build` directory to preview the page/s you've edited. 
 
 The build process may generate this error: ``WARNING: toctree contains reference to document u'foo' that doesn't have a title: no link will be generated``. It can be ignored as it does not negatively impact the documentation. 
+
+## Translating
```
Comment (jasonblais) [inline README.md:59]: Thank you for including this! Small proposed updates:

```
## Translating

1. Install [sphinx-intl](https://pypi.org/project/sphinx-intl/): `pip install sphinx-intl`.
2. `cd` into the cloned repository.
3. Install the required packages: `pipenv install`.
4. Extract translatable messages into .pot files: `make gettext`.
5. Generate .po files: `sphinx-intl update -p build/locale -l <your-language>`.
6. Translate .po files in `source/locale/<your-language>/LC_MESSAGES`. The translations should be added into ‘msgstr’ for each msgid.
7. Build translated files: `make -e SPHINXOPTS="-D language='<your-language>'" html`.

where `<your-language>` refers to the two-letter code of your language, such as `it` for Italian or `jp` for Japanese.

For more information please refer to [https://www.sphinx-doc.org/en/master/usage/advanced/intl.html](https://www.sphinx-doc.org/en/master/usage/advanced/intl.html)
```

**Comment by jasonblais** [comment]
@ctlaltdieliet, @der-test, @wget, @kaakaa, @flynbit, @uhlhosting: this PR adds support for translating product documentation, with the Getting Started guide translated in Italian.

@sandrospadaro has been doing fantastic work on it since March. We also have a group in Japan who is looking to submit some documentation translations for Japanese, using this same approach.

Completely optional, but if you have any feedback, would love to hear. Current status with outstanding items to review is summarized here: https://github.com/mattermost/docs/pull/3457#pullrequestreview-404424503

----

@sandrospadaro: @ctlaltdieliet, @der-test, @wget, @kaakaa, @flynbit, @uhlhosting help with Mattermost UI translations for Dutch, German, French, Japanese, Romanian respectively.

**Comment by wget** [comment]
@jasonblais Have browsed the discussion, but if this is to do things manually like taking a .pot (which will digress over time without any resync process in place) and writing the translation manually with a .po editor, that's a big no for me.

Instead of adding workload to the translation community, we are first looking forwards to fix the existent.

**Comment by sandrospadaro** [comment]
@jasonblais 

About ".readthoedocs.yml": I had add it to manage dependencies in readthedocs.
About different languages: You can add more translation projects to the main project in readtherocs. This will produce the related option.
About screenshot: there aren't mattermost applications translated in Italian, or i didn't find them. So i belive that is better to keep the original English versions of the screenshots.

Thanks

**Comment by jasonblais** [comment]
@wget:

> Have browsed the discussion, but if this is to do things manually like taking a .pot (which will digress over time without any resync process in place) and writing the translation manually with a .po editor, that's a big no for me. (Unless I'm wrong and you explain to me what's need to be done).

Is the concern related to "6. To maintain high standards for translated documentation, set up a process to notify translators when something is changed in the English-version of the docs, and expectation on correcting translations for new releases"

> Instead of adding workload to the translation community, we are first looking forwards to fix the existent.

By not adding this process to the translations server - at least until we've resolved the existing challenges, this shouldn't affect the workload on existing translations contributors.

**Comment by jasonblais** [comment]
@wget Let me know if you have any thoughts on my latest response :)

**Comment by wget** [comment]
> Is the concern related to "6. To maintain high standards for translated documentation, set up a process to notify translators when something is changed in the English-version of the docs, and expectation on correcting translations for new releases"

It isn't. I mean that I have been in a position to be in charge of several translation projects and playing around with .pot and .po files manually without any **authoritative server** in place (i.e. what's happening with the FileZilla project) has always led to a nightmare. Just have a look at how the situation appears with this huge pull request. And it is only for one language...

> By not adding this process to the translations server - at least until we've resolved the existing challenges, this shouldn't affect the workload on existing translations contributors.

It actually is. Because the translators that would be in charge of the user guide will be the same persons as the ones for the translations. Let's focus on the challenges we have for the translation process and let's add this new users guides requirement to the feedback translation process.

I know having the user guides translated is a requirement for some markets, and the new partnership I have been advocating last week showcased this problem.

In French, we have an expression for the current situation: "courir plusieurs lapins à la fois" ("running against several rabbits at once") translated as the nearest expression: "jumping from one boat to another".
 
If you authorize to have translations for the user guide now *without any authoritative server*, this will be a nightmare to maintain even on the short term. Let's wait for the new translation process being in place. It is now my priority at Mattermost.

The soonest Weblate is installed the soonest we will have user guides n the translation process =)

**Comment by jasonblais** [comment]
Thanks William. I'm open to keeping this open while we figure out the translations server,

> Because the translators that would be in charge of the user guide will be the same persons as the ones for the translations.

Short answer is no. I'm not looking to add user guides as a requirement for the existing translators.

**Comment by jasonblais** [comment]
@wget To summarize your feedback, is your suggestion to first do the product translations migration to Weblate (solving painpoints in product translations), then sync the docs in Weblate as well where the docs can be translated?

**Comment by jasonblais** [comment]
FYI that I've shared the same questions from above with the Japanese translated docs: https://github.com/mattermost/docs/pull/3664

(except those already resolved above)

**Comment by jasonblais** [comment]
@justinegeffen any updates for this PR? Thanks! :)

**Comment by justinegeffen** [comment]
Thanks for the ping, @jasonblais. Will revert shortly.

**Comment by justinegeffen** [comment]
@sandrospadaro, thank you so much for this PR. When this PR was submitted I didn't have as much time available as I needed to review this. Since then our docs team has grown, and we've also revised our content radically. I really do apologize for the delay as well as the change in documentation circumstances.

As @wget has commented on this thread, there are process and tool considerations to bear in mind and I'd like to loop @cwarnermm into this PR for visibility as well. :)

**Pull request closed without merging**